packages/components/filter-bar/pipes/pipe-text.ts

```diff
@@ -0,0 +1,88 @@
+import { NgClass } from '@angular/common';
+import { AfterViewInit, ChangeDetectionStrategy, Component, OnInit, ViewChild, ViewEncapsulation } from '@angular/core';
+import { FormControl, FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { ENTER } from '@koobiq/cdk/keycodes';
+import { KbqButtonModule } from '@koobiq/components/button';
+import { PopUpPlacements } from '@koobiq/components/core';
+import { KbqDividerModule } from '@koobiq/components/divider';
+import { KbqFormFieldModule } from '@koobiq/components/form-field';
+import { KbqInputModule } from '@koobiq/components/input';
+import { KbqPopoverModule, KbqPopoverTrigger } from '@koobiq/components/popover';
+import { KbqTextareaModule } from '@koobiq/components/textarea';
+import { KbqTitleModule } from '@koobiq/components/title';
+import { KbqBasePipe, KbqPipeMinWidth } from './base-pipe';
+import { KbqPipeButton } from './pipe-button';
+import { KbqPipeState } from './pipe-state';
+import { KbqPipeTitleDirective } from './pipe-title';
+
+@Component({
+    standalone: true,
+    selector: 'kbq-pipe-text',
+    templateUrl: 'pipe-text.html',
+    styleUrls: ['base-pipe.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+    providers: [
+        {
+            provide: KbqBasePipe,
+            useExisting: this
```

Comment:
что содержится в `this` в момент определения провайдера? 
```suggestion
            useExisting: KbqPipeTextComponent
```

Reply:
Это же 'красивый' способ "замкнуть" провайдинг чего либо на сам компонент :)

На самом деле не знаю могут быть какие либо проблемы с таким определением или нет, пока оставлю и понаблюдаю если будут ошибки изменю.

Reply:
я посмотрел доку и использование в material нигде не пишут `this`, а явно указывают ссылку на класс, боюсь, что в будущем может словить неприятный баг...